### PR TITLE
Update Node dependencies for SignalR

### PR DESCRIPTION
**Pull request by BrennanConroy**
Converted ConnectionTests.ts to have async tests, somehow using the `done` pattern caused parallelization issues after updating deps, and anyways, async is better.

### CODE REVIEW

File: src/SignalR/clients/ts/signalr/src/HubConnection.ts [hunk 1]
```diff
@@ -720,7 +720,7 @@ export class HubConnection {
             this.logger.log(LogLevel.Information, "Connection reconnecting.");
         }
 
-        if (this.onreconnecting) {
+        if (this.reconnectingCallbacks.length !== 0) {
```
Comment (halter73) [inline src/SignalR/clients/ts/signalr/src/HubConnection.ts:723]: Oops. 😆

**Review by halter73 — APPROVED** [review]

**Comment by halter73** [comment]
> Converted ConnectionTests.ts to have async tests, somehow using the done pattern caused parallelization issues after updating deps, and anyways, async is better.

Do you have more details on this? While I agree async is better, it would be nice to understand the issue was with the done pattern to make sure there isn't a similar issue with async.

**Comment by BrennanConroy** [comment]
> Do you have more details on this?

Unfortunately I couldn't find anything. I tried looking at the changes between the two versions and couldn't see anything super obvious. There was a change to allow parallel test execution, but it looked like it got reverted, and it was behind a flag I think.
I did see something about the `done` pattern either being obsoleted or suggested that it not be used.

**Comment by halter73** [comment]
Do you have any example failures that you can link to? Do you know if there could be problems for customers that might be using `done` instead of async/await?

**Comment by BrennanConroy** [comment]
> Do you have any example failures that you can link to?

No, it was all local test failures before making a PR. Basically the test logs were interleaved and some checks were looking for specific logs to exist or not exist and were failing because of that.

> Do you know if there could be problems for customers that might be using `done` instead of async/await?

No clue, but we don't ship the testing framework so I'm not worried about customers hitting this.

**Pull request merged**